### PR TITLE
Add soap action in the transaction name

### LINKED ISSUE #637: Using APM with SOAP
I would like to use APM in a SOAP web service project. I succeeded to use ElasticApmModule based on the sample "AspNetFullFrameworkSampleApp" from the repository. 

Nevertheless, I'm not really happy because the SoapAction is added in the header of the request and it's not to use to build the transaction name. So, for a same path, we aggregate all the actions.

Ex: 
/Search.asmx 
-> Action GetCustomers
-> Action GetFamillies

Solution : 
Allow to extend the ElasticApmModule with a custom code to override the transaction name ?

### PR DESCRIPTION
**Pull request by dhtek**
Closes #637 

Retrieve the soap action from the header request only if it's available and add it to the transaction name.

Code has been implemented to work with soap 1.1 where soap action is present in the http header.

### CODE REVIEW

**Comment by elasticcla** [comment]
Hi @dhtek, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in your Git [commit](https://github.com/elastic/apm-agent-dotnet/pull/683.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

File: src/Elastic.Apm.AspNetFullFramework/ElasticApmModule.cs [hunk 1]
```diff
@@ -102,6 +103,14 @@ private void ProcessBeginRequest(object eventSender)
 			var httpApp = (HttpApplication)eventSender;
 			var httpRequest = httpApp.Context.Request;
 
+			var transactionName = $"{httpRequest.HttpMethod} {httpRequest.Path}";
+
+			var soapAction = httpRequest.ExtractSoapAction(_logger);
+			if (soapAction != null)
+			{
```
Comment (gregkalapos) [inline src/Elastic.Apm.AspNetFullFramework/ElasticApmModule.cs]: nit: please remove braces (this is from our `.editorconfig`)

File: src/Elastic.Apm.AspNetFullFramework/Extensions/RequestExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Text;
```
Comment (gregkalapos) [inline src/Elastic.Apm.AspNetFullFramework/Extensions/RequestExtensions.cs]: nit: unused

File: src/Elastic.Apm.AspNetFullFramework/Extensions/RequestExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		/// <param name="logger">The logger.</param>
+		public static string ExtractSoapAction(this HttpRequest request, IApmLogger logger)
+		{
+			try
+			{
+				var soapActionWithNamespace = request.Headers.Get(SoapActionHeaderName);
+
+				if (string.IsNullOrWhiteSpace(soapActionWithNamespace)) return null;
+
+				var indexPosition = soapActionWithNamespace.LastIndexOf(@"/");
```
Comment (gregkalapos) [inline src/Elastic.Apm.AspNetFullFramework/Extensions/RequestExtensions.cs]:
```suggestion
				var indexPosition = soapActionWithNamespace.LastIndexOf(@"/", StringComparison.InvariantCulture);
```

File: src/Elastic.Apm.AspNetFullFramework/Extensions/RequestExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+				if (indexPosition != -1)
+				{
```
Comment (gregkalapos) [inline src/Elastic.Apm.AspNetFullFramework/Extensions/RequestExtensions.cs]: nit: please remove `{` `}`

**Comment by dhtek** [comment]
> 
> 
> Hi @dhtek, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in your Git [commit](https://github.com/elastic/apm-agent-dotnet/pull/683.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

Yes it's done now, can you check if it's ok for you ?

**Comment by gregkalapos** [comment]
> > Hi @dhtek, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in your Git [commit](https://github.com/elastic/apm-agent-dotnet/pull/683.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?
> 
> Yes it's done now, can you check if it's ok for you ?

Hmm, unfortunately our CLA checker still says the PR does not have CLA signature. 

Did you use your mail address that you see here in the 2. line? https://patch-diff.githubusercontent.com/raw/elastic/apm-agent-dotnet/pull/683.patch

In the CLA database I only see your gmail address.

**Comment by robgil** [comment]
# [Codecov](https://codecov.io/gh/elastic/apm-agent-dotnet/pull/683?src=pr&el=h1) Report
> Merging [#683](https://codecov.io/gh/elastic/apm-agent-dotnet/pull/683?src=pr&el=desc) into [master](https://codecov.io/gh/elastic/apm-agent-dotnet/commit/6f7abfc5fe49f90473db6351a4f91d1311f40a52&el=desc) will **decrease** coverage by `0.18%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/elastic/apm-agent-dotnet/pull/683?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #683      +/-   ##
==========================================
- Coverage   74.78%   74.60%   -0.19%     
==========================================
  Files         113      113              
  Lines        4962     4962              
==========================================
- Hits         3711     3702       -9     
- Misses       1251     1260       +9     
```


| [Impacted Files](https://codecov.io/gh/elastic/apm-agent-dotnet/pull/683?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [...c.Apm/Metrics/MetricsProvider/GcMetricsProvider.cs](https://codecov.io/gh/elastic/apm-agent-dotnet/pull/683/diff?src=pr&el=tree#diff-c3JjL0VsYXN0aWMuQXBtL01ldHJpY3MvTWV0cmljc1Byb3ZpZGVyL0djTWV0cmljc1Byb3ZpZGVyLmNz) | `63.01% <0.00%> (-6.17%)` | :arrow_down: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/elastic/apm-agent-dotnet/pull/683?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/elastic/apm-agent-dotnet/pull/683?src=pr&el=footer). Last update [6f7abfc...f51eaba](https://codecov.io/gh/elastic/apm-agent-dotnet/pull/683?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Pull request merged**